src/axom/core/Array.hpp

```diff
@@ -642,10 +663,13 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM, SPACE>>
    * \note If used in a device kernel, the number of push_backs must not exceed
    *  the capacity, since device-side reallocations aren't supported.
    * \note Array must be allocated in unified memory if calling on the device.
-   * 
+   *
    * \pre DIM == 1
    */
-  AXOM_HOST_DEVICE void push_back(T&& value);
+  /// @{
+  AXOM_HOST_DEVICE void pushBackDevice(const T& value);
+  AXOM_HOST_DEVICE void pushBackDevice(T&& value);
```

Comment:
For consistency, should these be `push_back_device()` and `emplace_back_device()`?

Reply:
I wondered that too but didn't have too strong an opinion.

Reply:
Why are these methods host-device decorated? Having "Device" in their names implies that they should only be called in a device context, at least that's my interpretation.  Am I missing something?